action.go

```diff
@@ -35,8 +44,9 @@ type Action interface {
 	PreMachine(context *DebosContext, m *fakemachine.Machine, args *[]string) error
 	PreNoMachine(context *DebosContext) error
 	Run(context *DebosContext) error
-	Cleanup(context DebosContext) error
-	PostMachine(context DebosContext) error
+	Cleanup(context *DebosContext) error
+	PostCleanup(context *DebosContext) error
```

Comment:
If you were to split out the passing by reference of all the things in a sepeate commit this would have been so so much easier to review